src/main/java/org/opensrp/repository/postgres/mapper/custom/xml/CustomLocationMetadataMapper.xml

```diff
@@ -179,29 +179,32 @@
 	<select id="selectLocationHierachy"
 			resultMap="DetailsResultMap">
 		select lm.geojson_id identifier ,lm."name" ,lm.parent_id, 'false' voided, lm.location_id,string_agg(lt."name",',') tags,
-		l.json->'properties'->>'geographicLevel' geographic_level
+		l.json->'properties'->>'geographicLevel' geographic_level,
+		record.level
 		from core.location_metadata lm
 		left join core.location_tag_map ltm on ltm.location_id =lm.location_id 
 		left join core.location_tag lt on lt.id =ltm.location_tag_id
```

Comment:
Introduce a flag to mark location tags optional, if they are not required grouping can be avoided